frontend/src/lib/components/LemonBubble/LemonBubble.tsx

```diff
@@ -4,31 +4,48 @@ import { CSSTransition } from 'react-transition-group'
 import './LemonBubble.scss'
 
 export interface LemonBubbleProps {
-    count?: number
+    count?: number | JSX.Element
     size?: 'small' | 'medium' | 'large'
     position?: 'none' | 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right'
     showZero?: boolean
+    borderless?: boolean
+    className?: string
+    style?: React.CSSProperties
 }
 
 /** An icon-sized Bubble for displaying a count.
  *
- *  Numbers up to 9 are displayed in full, in integer form, with 9+ for higher values
+ * Numbers up to 9 are displayed in full, in integer form, with 9+ for higher values.
+ * JSX elements are rendered outright to support use cases where the badge is meant to show an icon.
+ * If `showZero` is set to `true`, the component won't be hidden if the count is 0.
  */
 export function LemonBubble({
     count,
     size = 'medium',
     position = 'none',
     showZero = false,
+    className,
+    ...spanProps
 }: LemonBubbleProps): JSX.Element {
     // NOTE: We use 1 for the text if not showing so the fade out animation looks right
-    const text = typeof count === 'number' && count !== 0 ? (count < 10 ? String(count) : '9+') : showZero ? '0' : '1'
+    const text =
```

Comment:
Generally I'm a little skeptical of the maintainability of something like this... Feels like there would be a lot of cases where this won't work and only works for this feature. It makes it easy for someone to accidentally code a bad LemonBubble which is the opposite of what the Design Systems aims are.

Just sharing thoughts though, not sure what the right approach is

Reply:
I feel like maybe a `children`-based API would work better here, because according to the design system, the badge component definitely should support icons (e.g. [here](https://www.figma.com/file/Y9G24U4r04nEjIDGIEGuKI/PostHog-Design-System-One?node-id=3143%3A4058) it has an exclamation mark).

Reply:
Interesting - hadn't seen that section.

In which case I feel like we should have some more storybook entries so we can visually be sure that icons will be rendered consistently as they should be. Doesn't have to be done now of course but at some point, otherwise we will end up with a lot of custom css going around to hack icons to fit in there properly.